sources/logdog/src/main.rs

```diff
@@ -0,0 +1,218 @@
+use create_tarball::create_tarball;
+use error::Result;
+use tempfile::TempDir;
+
+use snafu::{ErrorCompat, ResultExt};
+
+const ERROR_FILENAME: &str = "logdog.errors";
+const OUTPUT_FILENAME: &str = "bottlerocket-logs.tar.gz";
+const TARBALL_DIRNAME: &str = "bottlerocket-logs";
+
+/// Prints a usage message in the event a bad arg is passed.
+fn usage() -> ! {
+    let program_name = env::args().next().unwrap_or_else(|| "program".to_string());
+    eprintln!(
+        r"Usage: {}
+            [ --output PATH ]       where to write archived logs
+",
+        program_name,
+    );
+    process::exit(2);
+}
+
+/// Prints a more specific message before exiting through usage().
+fn usage_msg(msg: &str) -> ! {
+    eprintln!("{}\n", msg);
+    usage();
+}
+
+/// Parses the command line arguments.
+fn parse_args(args: env::Args) -> PathBuf {
+    let mut output_arg = None;
+    let mut iter = args.skip(1);
+    while let Some(arg) = iter.next() {
+        match arg.as_ref() {
+            "--output" => {
+                output_arg = Some(
+                    iter.next()
+                        .unwrap_or_else(|| usage_msg("Did not give argument to --output")),
+                )
+            }
+            _ => usage(),
+        }
+    }
+
+    match output_arg {
+        Some(path) => PathBuf::from(path),
+        None => env::temp_dir().as_path().join(OUTPUT_FILENAME),
+    }
+}
+
+/// Runs a command and writes its output to a file.
+pub(crate) fn run_command<P: AsRef<Path>>(output_filepath: P, command: &str) -> Result<()> {
+    let command_parts: Vec<String> = command
+        .to_owned()
+        .split(" ")
+        .map(|s| s.to_string())
+        .collect::<Vec<String>>();
+    let command = match command_parts.get(0) {
+        Some(c) => c.into(),
+        None => "".to_string(),
+    };
+    let args: Vec<String> = if command_parts.len() > 1 {
+        command_parts[1..].to_owned()
+    } else {
+        vec![]
+    };
+    let ofile = File::create(output_filepath.as_ref()).context(error::CommandOutputFile {
+        path: output_filepath.as_ref(),
+    })?;
+    let stderr_file = ofile.try_clone().context(error::CommandErrFile {
+        path: output_filepath.as_ref(),
+    })?;
+    Command::new(command.as_str())
+        .args(&args)
+        .stdout(Stdio::from(ofile))
+        .stderr(Stdio::from(stderr_file))
+        .spawn()
+        .context(error::CommandSpawn {
+            command: command.clone(),
+        })?
+        .wait_with_output()
+        .context(error::CommandFinish {
+            command: command.clone(),
+        })?;
+    Ok(())
+}
+
+/// Runs a list of commands and writes all of their output into files in the same `outdir`.  Any
+/// failures are noted in the file named by ERROR_FILENAME.  This function ignores the commands'
+/// return status and only fails if we can't save our own errors. The commands are specified by
+/// tuples where `.0` is the desired output filename and `.1` is the command to run.
```

Comment:
Perhaps a `struct LogRequest<'a, 'b> { filename: &'a str, command: &'b str }` would help, in terms of naming these things and removing the tuple syntax?